ApplicationCode/Application/RiaPreferences.cpp

```diff
@@ -405,6 +418,42 @@ RiaPreferences* RiaPreferences::current()
     return RiaApplication::instance()->preferences();
 }
 
+// Validates a wild card pattern (simpler for user)
+// Has to use the older QRegExp because QRegularExpression didn't add any
+// support for wild cards until Qt 5.12.
+class ValidRegExpValidator : public QValidator
+{
+public:
+    ValidRegExpValidator( const QString& defaultPattern )
```

Comment:
Consider moving this class into a separate file as RiaPreferences is quite large

Reply:
done